src/pages/batchImports/[resourceId]/index.js

```diff
@@ -0,0 +1,296 @@
+import Table from 'src/components/Shared/Table'
+import { useContext, useEffect, useState } from 'react'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Button, Grid } from '@mui/material'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { ControlContext } from 'src/providers/ControlContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import toast from 'react-hot-toast'
+import { formatDate, formatDateDefault, formatDateForImport } from 'src/lib/date-helper'
+import { ProgressForm } from 'src/components/Shared/ThreadProgress'
+import { useWindow } from 'src/windows'
+import { useError } from 'src/error'
+import WindowToolbar from 'src/components/Shared/WindowToolbar'
+
+const transform = array => {
+  return {
+    count: array?.length > 0 ? array.length : 0,
+    list:
+      array?.length > 0
+        ? array.map((item, index) => ({
+            ...item,
+            recordId: index + 1,
+            minPrice: item.minPrice || 0,
+            maxAccess: item.mandatory
+          }))
+        : [],
+    statusId: 1,
+    message: '',
+    _startAt: 0
+  }
+}
+
+const convertValue = (value, dataType, isAPI = false) => {
+  if (value === '') {
+    return value
+  }
+  switch (dataType) {
+    case 2:
+      return parseInt(value, 10)
+    case 3:
+      return parseFloat(value)
+    case 5:
+      return isAPI ? formatDateForImport(value) : formatDateDefault(formatDate(value))
+    default:
+      return value
+  }
+}
+
+const parseCSV = (text, columns) => {
+  const lines = text.split('\n').filter(line => line.trim())
+
+  if (lines.length === 0) return
+
+  const headers = lines[0].split(',').map(header => header.trim())
+
+  const columnMap = columns.reduce((map, col) => {
+    map[col.headerName] = col
+
+    return map
+  }, {})
+
+  const orderedColumns = headers.map(header => columnMap[header])
+
+  const rows = lines.slice(1).map(line => {
+    const values = line.split(',').map(value => value.trim())
+
+    return orderedColumns.reduce((obj, col, index) => {
+      const header = headers[index]
+      if (col) {
+        obj[col.field] = convertValue(values[index], col.dataType)
+      } else {
+        obj[header] = values[index]
+      }
+
+      return obj
+    }, {})
+  })
+
+  return transform(rows)
+}
+
+const getImportData = (gridData, columns, stackError) => {
+  const mandatoryColumns = columns.filter(col => col.mandatory)
+
+  const missingFields = gridData.list.flatMap(row =>
+    mandatoryColumns
+      .filter(col => row[col.field] === null || row[col.field] === undefined || row[col.field] === '')
+      .map(col => col.headerName)
+  )
+
+  if (missingFields.length > 0) {
+    const uniqueMissingFields = [...new Set(missingFields)]
+    stackError({
```

Comment:
Throw the message out in a throw new Error('something'). handle the error inside the component and there you should stack error.